sentry-kotlin-compiler-plugin/src/main/kotlin/io/sentry/compose/JetpackComposeTracingIrExtension.kt

```diff
@@ -0,0 +1,270 @@
+                                val sentryModifier = irTemporary(
+                                    irCall(sentryModifierFunctionRef, modifierType).also { call ->
+                                        call.extensionReceiver =
+                                            irGetObject(modifierCompanionClassRef)
+                                        call.putValueArgument(0, irString(name))
+                                    },
+                                    nameHint = SENTRY_BASE_MODIFIER
+                                )
+
+                                visitingFunctionSentryModifier.add(irGet(sentryModifier))
+                                visitingFunctionNames.add(name)
+                                modifierAdded = true
+
+                                for (statement in body.statements) {
+                                    +statement
+                                }
+                            }
+                    }
+                }
+                // in case we didn't add a modifier, add an empty modifier to keep the stack in sync
+                if (!modifierAdded) {
+                    visitingFunctionSentryModifier.add(null)
+                    visitingFunctionNames.add(name)
+                }
+                val irStatement = super.visitFunctionNew(declaration)
+
+                visitingFunctionNames.removeLast()
+                visitingFunctionSentryModifier.removeLast()
+
+                return irStatement
+            }
+
+            override fun visitCall(expression: IrCall): IrExpression {
+                // Case A: modifier is not supplied
+                // -> simply set our modifier as param
+                // e.g. BasicText(text = "abc")
+                // into BasicText(text = "abc", modifier = sentryBaseModifier)
+                val modifierArgumentIndex = expression.symbol.owner.valueParameters.indexOfFirst {
+                    it.type.classFqName == modifierClassFqName
+                }
+
+                if (modifierArgumentIndex != -1) {
+                    val modifierArgument: IrExpression? =
+                        expression.getValueArgument(modifierArgumentIndex)
+
+                    // we can safely set the sentryModifier if there's no value parameter provided
+                    // but in case the Jetpack Compose Compiler plugin runs before us,
+                    // it will inject all default value parameters as actual parameters using IrComposite
+                    // hence we need to cover this case and overwrite the composite default/null value with sentryModifier
+                    val setModifier = (
+                        modifierArgument == null ||
+                            (
+                                modifierArgument is IrComposite &&
+                                    modifierArgument.origin == IrStatementOrigin.DEFAULT_VALUE &&
+                                    modifierArgument.type.classFqName == kotlinNothing
```

Comment:
`l`: maybe a small code snippet as comment would help to understand what are we trying to cover here, because I was confused by `kotlinNothing`